sdk/src/test/scala/io/horizen/account/api/rpc/service/EthServiceTest.scala

```diff
@@ -1329,4 +1329,25 @@ class EthServiceTest extends JUnitSuite with MockitoSugar with ReceiptFixture wi
     assertJsonEquals(txPoolInspectOutput, ethService.execute(getRpcRequest()))
   }
 
+  @Test
+  def eth_getSHA3(): Unit = {
+    val method = "web3_sha3"
+
+    val validCases = Table(
+      ("Input", "Expected output"),
+      ("0x68656c6c6f20776f726c64", "0x47173285a8d7341e5e972fc677286384f802f8ef42a5ec5f03bbfa254cb01fad"),
+      ("68656c6c6f20776f726c64", "0x47173285a8d7341e5e972fc677286384f802f8ef42a5ec5f03bbfa254cb01fad"),
+      ("ZZXX68656c6c6f20776f726c64", "0x8bc0488d7a81d4c07855cf17f63c41b5f6cc6c0d15c764ce43ab668c9e6809ba"),
+      ("0x", "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"),
+      ("", "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470"),
+      ("text", "0x2c7a9a0b269b5b740e242917d5b704ce4329a174526cd76ba1f042dfd88795bb")
+    )
+
+    forAll(validCases) { (input, expectedOutput) =>
+      assert(
+        expectedOutput == ethService.execute(getRpcRequest(paramValues = Array(input), method = method))
```

Comment:
Ok, now I see the issue. You need to merge in `dev`, there have been a lot of changes in the RPC api  and its tests.

This line here should simplify to: `assertJsonEquals(expectedOutput, rpc("web3_sha3", input))` and you need to updated the expected output to include the quotation marks, i.e. `"\"0x0000000000000000000000000000000000000000000000000000000000000000\""`

Reply:
After merging `dev` all cases without the `0x` prefix will fail. We could use some of them as tests for invalid cases.

Reply:
Thanks for the info and clearing that up! After the merge with `dev`, the tests are modified and work as expected.